tests/test_crypto.py

```diff
@@ -284,3 +285,52 @@ def test_reencrypt_secrets_and_write_to_yaml_file():
     # Test the value is re-encrypted 
     test_encrypted_key_value = decrypt_value(after_dict[test_encrypted_key], private_key_new)
     assert test_encrypted_key_value == "C"
+
+
+def test_add_nested_secret_to_yaml_file():
+    private_key_output_filename = "test_output/test_add_nested_secret_to_yaml_file.private"
+    public_key_output_filename = "test_output/test_add_nested_secret_to_yaml_file.public"
+    private_key = generate_private_key_to_file(private_key_output_filename)
+    public_key = generate_public_key_to_file(private_key_output_filename, public_key_output_filename)
+    test_key_value = "TEST_PARENT_KEY"
+    test_nested_key_value = "TEST_PARENT_KEY:TEST_CHILD_KEY"
+
+    yaml_file_fixture = "fixtures/test_add_nested_secret_to_yaml_file.yml"
+    yaml_file_to_append_to = "test_output/test_add_nested_secret_to_yaml_file.yml"
+    copyfile(yaml_file_fixture, yaml_file_to_append_to)
+
+    before_dict = None
+    with open(yaml_file_to_append_to, "r") as f:
+        before_dict = load(f)
+
+    # Check for expected test data
+    assert before_dict["Y"] == "B"
+    assert before_dict["X"] == "B"
+
+    # Check the value we are adding is not already there
+    add_secret_to_yaml_file(test_nested_key_value, test_key_value, public_key_output_filename, yaml_file_to_append_to)
+
+    with open(yaml_file_to_append_to, "r") as f:
+        after_dict = load(f)
+    # Test the value was added
+    assert test_key_value in after_dict
+
+    # Test the value is encrypted.
+    assert isinstance(reduce(lambda x, y: x[y], test_nested_key_value.split(":"), after_dict), Encrypted)
+
+    # Test the value was actually encrypted
+    assert reduce(lambda x, y: x[y], test_nested_key_value.split(":"), after_dict) != test_key_value
```

Comment:
This is currently testing that the nested value is not "TEST_PARENT_KEY"
it should be testing that it is not equal to "test" I believe.